examples/v1/workflows/durable-sleep.go

```diff
@@ -0,0 +1,53 @@
+package v1_workflows
+
+import (
+	"strings"
+	"time"
+
+	v1 "github.com/hatchet-dev/hatchet/pkg/v1"
+	"github.com/hatchet-dev/hatchet/pkg/v1/task"
+	"github.com/hatchet-dev/hatchet/pkg/v1/workflow"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+type DurableSleepInput struct {
+	Message string
+}
+
+type SleepOutput struct {
+	TransformedMessage string
+}
+
+type DurableSleepOutput struct {
+	Sleep SleepOutput
+}
+
+func DurableSleep(hatchet *v1.HatchetClient) workflow.WorkflowDeclaration[DurableSleepInput, DurableSleepOutput] {
+
+	simple := v1.WorkflowFactory[DurableSleepInput, DurableSleepOutput](
+		workflow.CreateOpts[DurableSleepInput]{
+			Name: "durable-sleep",
+		},
+		hatchet,
+	)
+
+	simple.DurableTask(
+		task.CreateOpts[DurableSleepInput]{
+			Name: "Sleep",
+			Fn: func(input DurableSleepInput, ctx worker.DurableHatchetContext) (*SleepOutput, error) {
+
+				_, err := ctx.SleepFor(time.Minute)
```

Comment:
nit: in Python I'm calling this `wait_for` - do you think we should be consistent here? or is this just a wrapper around `wait_for`? if it is, maybe I should implement that too